bundles/org.openhab.binding.bmwconnecteddrive/README.md

```diff
@@ -0,0 +1,686 @@
+Basically 3 types of information are registered as properties
+
+* Informations regarding your dealer with address and phone number
+* Which services are available / not available
+* Vehicle properties like color, model type, drive train and construction year
+
+In the right picture can see in *Services Activated* e.g. the *DoorLock* and *DoorUnlock* services are mentioned. 
+This ensures channel group [Remote Services](#remote-services) is supporting door lock and unlock remote control.
+
+In  *Services Supported* the entry *LastDestination* is mentioned.
+So it's valid to connect channel group [Last Destinations](#destinations) in order to display and select the last navigation destinations.
+
+| Property Key       | Property Value      |  Supported Channel Groups    |
+|--------------------|---------------------|------------------------------|
+| Services Supported | Statistics          | last-trip, lifetime          |
+| Services Supported | LastDestinations    | destinations                 |
+| Services Activated | _list of services_  | remote                       |
+
+
+## Discovery
+
+Auto discovery is starting after the bridge towards BMW's ConnectedDrive is created. 
+A list of your registered vehicles is queried and all found things are added in the inbox.
+Unique identifier is the *Vehicle Identification Number* (VIN). 
+If a thing is already declared in a  _.things_ configuration, discovery won't highlight it again.
+Properties will be attached to predefined vehicles if the VIN is matching.
+
+## Configuration
+
+### Bridge Configuration
+
+| Parameter       | Type    | Description                                                        |           
+|-----------------|---------|--------------------------------------------------------------------|
+| userName        | text    | BMW ConnectedDrive Username                                       |
+| password        | text    | BMW ConnectedDrive Password                                       |
+| region          | text    | Select region in order to connect to the appropriate BMW server.   |
+
+The region Configuration has 3 different options
+
+* _NORTH_AMERICA_
+* _CHINA_
+* _ROW_  (Rest of World)
+
+### Thing Configuration
+
+Same configuration is needed for all things
+
+| Parameter       | Type    | Description                           |           
+|-----------------|---------|---------------------------------------|
+| vin             | text    | Vehicle Identification Number (VIN)   |
+| refreshInterval | integer | Refresh Interval in Minutes           |
+| units           | text    | Unit Selection. See below.            |
+| imageSize       | integer | Image Size                            |
+| imageViewport   | text    | Image Viewport                        |
+
+The unit configuration has 3 options
+
+* _AUTODETECT_ selects miles for US & UK, kilometer otherwise
+* _METRIC_ selects directly kilometers
+* _IMPERIAL_ selects directly miles
```

Comment:
Is this setting used for the representation within OH or for intepreting the API calls?

Reply:
It's used to pick the right element from json response.

```
...
"remainingRangeElectric": 184,
"remainingRangeElectricMls": 114,
...
```

Reply:
Why not picking always the same item and use QuantityType to convert it? Then, the user can rely on his system settings and this setting would be unnecessary.

Reply:
In principle I would say yes. I think this would satisy more than 99% of the users. Nevertheless there are[ some countries besides UK & US using miles](https://en.wikipedia.org/wiki/Miles_per_hour). I refused to maintain a list locales for all these countries in this particular binding to have 100% accuracy. Therefore I came up with this: _Autodetect_ as default config will satisfy 99,9% of the users and in case it's wrong the user can select it by hard as a _last resort_.

Reply:
Isn't the API for all countries the same? I.e. both variants are present? Why not use one unit in your code and make a universal QuantityType, that the user can choose the unit himself/herself by selecting the right unit in the state pattern?

Reply:
API is the same but not for all values both values are given
* range values have separate json fields for mi and km
* trip values are alway given in km without additional mi field
* mileage - either km or mi, cannot be evaluted by the binding itsself. It's either 1234 km or 1234 mi without conversion of the value

I don't know if I fully got your idea. So you propose to define a combined km/mi QuantityType and the user needs to select the unit via format?

Reply:
Yes, that way you could remove the configuration options and the user can set the desired unit via the well-known pattern. Same as for temperatures to select Celsius/Fahrenheit.

Reply:
I see your point. Unfortunately it's not possible inside the binding if the _mileage_ integer value is referring to kilometers or miles. This is an internal car coding which needs a corresponding configuration by the user and cannot be evaluated by the binding itsself.